src/ngraph/builder/split.hpp

```diff
@@ -0,0 +1,54 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+#include <memory>
+
+#include "ngraph/node.hpp"
+
+namespace ngraph
+{
+    namespace builder
+    {
+        /// \brief      Split node on specified axis into multiple parts.
+        ///
+        /// \param[in]  node          The input node.
+        /// \param[in]  length_parts  The vector defining the lengts of each splitted part.
+        /// \param[in]  axis          The axis we split input node on. Default value is zero axis.
+        ///
+        /// \return     The vector containing multiple nodes we split input node into.
+        ///
+        NodeVector split(const std::shared_ptr<ngraph::Node>& node,
+                         const std::vector<std::size_t>& length_parts,
+                         std::size_t axis = 0);
+
+        /// \brief      Split node on specified axis into multiple parts.
+        ///
+        /// \param[in]  node          The input node.
+        /// \param[in]  split_parts   The number of parts we want to split input node at given
+        ///                           axis. The length of the axis to split must be divisible by
+        ///                           this value.
+        /// \param[in]  axis          The axis we split input node on. Default value is zero axis.
+        ///
+        /// \note       This implementation supports negative `axis` values (similar to NumPy
+        ///             indexing).
```

Comment:
Please explain, that this means negative values are substracted from the highest axis value (rank) of tensor.